### PR TITLE
Add support to install OctopusCLI based on wildcards

### PR DESCRIPTION
**Pull request by johnsimons**
Fixes https://github.com/OctopusDeploy/OctoTFS/issues/229

### CODE REVIEW

File: source/tasks/OctoInstaller/OctoInstallerV5/octopusCLIVersionFetcher.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { maxSatisfying, valid } from "semver";
+
+export class OctopusCLIVersionFetcher {
+    constructor(readonly versions: string[]) {}
+
+    public getVersion(versionSpec: string) {
+        if (versionSpec === "*") {
```
Comment (slewis74) [inline source/tasks/OctoInstaller/OctoInstallerV5/octopusCLIVersionFetcher.ts:7]: I think we should make "latest" a synonym for "*"

**Review by slewis74 — CHANGES_REQUESTED** [review]
I think we should make `latest` a synonym for `*`. I updated a v4 step and left the text and didn't get any failures on save. I tried to run and couldn't get our test instance working, but I'm pretty sure it won't do what I expected.

File: source/tasks/OctoInstaller/OctoInstallerV5/octopusCLIVersionFetcher.ts (hunk 1 above)
Comment (johnsimons) [reply source/tasks/OctoInstaller/OctoInstallerV5/octopusCLIVersionFetcher.ts:7]: I understand we did called it `latest` before, but not sure we should

Comment (slewis74) [reply source/tasks/OctoInstaller/OctoInstallerV5/octopusCLIVersionFetcher.ts:7]: The UX here is my concern. If I have a step with v4 selected and then switch to v5, my expectation would be that `latest` should either be rejected by the save or should be accepted to mean what it previously meant. Yaml can't support the idea of stopping the update based on validation, which is why I was leaning towards continuing to support it.
I couldn't get our test pipeline to run, what does the user see if `latest` is used?

Comment (johnsimons) [reply source/tasks/OctoInstaller/OctoInstallerV5/octopusCLIVersionFetcher.ts:7]: They will see an error, see https://github.com/OctopusDeploy/OctoTFS/pull/230/files#diff-61116d68378cf54c8ea8621ba761b95a9b1340f0cff243c15504b600f1501968R20

Comment (slewis74) [reply source/tasks/OctoInstaller/OctoInstallerV5/octopusCLIVersionFetcher.ts:7]: fair enough.

**Review by slewis74 — APPROVED** [review]

**Pull request merged**